File: Sorting Technique/Bubble_Sort.cpp
#include<bits/stdc++.h>
using namespace std;

void swap(int *xp, int *yp)
{
	int temp = *xp;
	*xp = *yp;
	*yp = temp;
}

void selectionSort(int arr[], int n)
{
	int i, j, min_indx;
	for(i=0; i<n-1; i++)
	{
		for(j=i+1; j<n; j++)
		{
			if(arr[j] < arr[min_indx])
			{
				swap(&arr[min_indx], &arr[i]);
			}
		}
	}
}

void printArray(int arr[], int size)
{
	int i;
	for(i=0; i<size; i++)
	{
		cout<<arr[i]<<" ";
	}
	cout<<endl;
}

int main()
{
	int i, n;
	cout<<"Enter Size: ";
	cin>>n;
	int arr[n];
	cout<<"\nEnter Array: ";
	for(i=0; i<n; i++)
	{
		cin>>arr[i];
	}
	selectionSort(arr, n);
	cout<<"\nArray after Sorting: ";
	printArray(arr, n);
	return 0;
}

/* Time Complexity: O(n*n) <Worst Case>
				 O(n) <Best Case when array is sorted>
Space Complexity: O(1) */

File: Sorting Technique/JAVA/BubbleSort.java
package sorting;

public class BubbleSort {

    public static void bubbleSort(int[] arr) {

        for (int i = 0; i < arr.length - 1; i++) {

            for (int j = 0; j < arr.length - i - 1; j++) {

                if(arr[j] > arr[j+1]) {

                    int temp = arr[j] ;
                    arr[j] = arr[j + 1] ;
                    arr[j+1] = temp ;
                }
            }
        }
    }

    public static void main(String[] args) {

        int[] arr = {64, 25, 12, 22, 11};

        System.out.print("The original array: ");
        for (int i : arr) {
            System.out.printf("%d ,", i);
        }
        System.out.println();

        bubbleSort(arr);

        System.out.print("The sorted array: ");
        for (int i : arr) {
            System.out.printf("%d ,", i);
        }
    }
}
